config/m4/ucx.m4

```diff
@@ -80,7 +95,17 @@ AS_IF([test "x$ucx_checked" != "xyes"],[
                 AC_SUBST(UCS_LDFLAGS, "-L$check_ucx_libdir")
             ])
 
-            AC_SUBST(UCX_LIBADD, "-lucp -lucs -lucm")
+            AS_IF([test "x$ucg_happy" = "xyes"],
+            [
+                ucg_major=$(cat $check_ucx_dir/include/ucg/api/ucg_version.h | grep -Po "UCG_API_MAJOR\s+\K\d+")
+                ucg_minor=$(cat $check_ucx_dir/include/ucg/api/ucg_version.h | grep -Po "UCG_API_MINOR\s+\K\d+")
+                AC_MSG_RESULT([Detected UCG version: ${ucg_major}.${ucg_minor}])
+                AC_SUBST(UCX_LIBADD, "-lucg -lucp -lucs -lucm")
```

Comment:
i don't think we want to add "-lucg" to global flags. It should be UCG_LIBADD parameter used only in CL/ucg. Main libucc.so should not depend on libucg (like it does not depend on say ibverbs, mcast or sharp).

Reply:
@vspetrov It's not added to global flags - only to UCX_LIBADD. This means every component using UCX will also use UCG (which is currently part of UCX, so it makes sense to me). I don't see which UCC component could use UCX without UCG (since UCG is not standalone, so it can't be a separate context anyway).

Reply:
no. For example, TL/UCX - low level component. It will be linked against UCP only. Why it should depend on UCG. I think you are mixing 2 independent things here. only install/lib/xccl/xccl_cl_ucg.so should depend on libucg.so. So, it makes sense actually to put libucg detection into xucg.m4 . First we run ucx.m4 - if it is detected, then we can run xucg.m4 and try detecting it. If it is also detected, then you populate UCG_CPPFLAGS, UCG_LDFLAGS, UCG_LIBADD and use them in the makefile for CL/ucg.